pkg/internal/vault/vault.go

```diff
@@ -279,6 +293,61 @@ func (v *Vault) requestTokenWithAppRoleRef(client Client, appRole *v1alpha1.Vaul
 	return token, nil
 }
 
+func (v *Vault) requestTokenWithKubernetesAuth(client Client, kubernetesAuth *v1alpha1.KubernetesAuth) (string, error) {
+	secret, err := v.secretsLister.Secrets(v.namespace).Get(kubernetesAuth.SecretRef.Name)
+	if err != nil {
+		return "", err
+	}
+
+	key := kubernetesAuth.SecretRef.Key
+	if key == "" {
+		key = "token"
```

Comment:
Please explicitly call out this default value in the comment/description for this SecretKeyRef - these defaults are near impossible to keep track of otherwise.